sync/class.jetpack-sync-module-full-sync.php

```diff
@@ -87,85 +81,127 @@ function start( $modules = null ) {
 
 	private function set_status_queuing_started() {
-		$status            = $this->initial_status;
-		$status['started'] = time();
-		$this->update_status( $status );
+		$this->clear_status();
+		$prefix = self::STATUS_OPTION_PREFIX;
+		$this->update_status_option( "started", time() );
 	}
 
 	private function set_status_queuing_finished() {
-		$this->update_status( array( 'queue_finished' => time() ) );
+		$prefix = self::STATUS_OPTION_PREFIX;
+		$this->update_status_option( "queue_finished", time() );
 	}
 
-	private $initial_status = array(
-		'started'        => null,
-		'queue_finished' => null,
-		'sent_started'   => null,
-		'finished'       => null,
-		'sent'           => array(),
-		'queue'          => array(),
-	);
+	private function get_status_option( $option ) {
+		$prefix = self::STATUS_OPTION_PREFIX;
+		$status = get_option( "{$prefix}_{$option}", null );
+		if ( is_null( $status ) ) {
+			return $status;
+		}
+		return intval( $status );
+	}
 
-	public function get_status() {
-		return get_option( self::STATUS_OPTION, $this->initial_status );
+	private function is_started() {
+		$prefix = self::STATUS_OPTION_PREFIX;
+		return ! is_null( get_option( "{$prefix}_started", null ) );
 	}
 
-	public function update_status( $status ) {
-		return update_option(
-			self::STATUS_OPTION,
-			array_merge( $this->get_status(), $status )
+	private function is_finished() {
+		$prefix = self::STATUS_OPTION_PREFIX;
+		return !! get_option( "{$prefix}_finished", null );
+	}
+
+	public function get_status() {
+		$prefix = self::STATUS_OPTION_PREFIX;
+		$status = array(
+			'started'        => $this->get_status_option( 'started' ),
+			'queue_finished' => $this->get_status_option( 'queue_finished' ),
+			'sent_started'   => $this->get_status_option( 'sent_started' ),
+			'finished'       => $this->get_status_option( 'finished' ),
+			'sent'           => array(),
+			'queue'          => array(),
 		);
+
+		foreach ( Jetpack_Sync_Modules::get_modules() as $module ) {
+			$queued = get_option( "{$prefix}_{$module->name()}_queued", null );
+			$sent = get_option( "{$prefix}_{$module->name()}_sent", null );
+
+			if ( ! is_null( $queued ) ) {
+				$status[ 'queue' ][ $module->name() ] = $queued;
+			}
+			
+			if ( ! is_null( $sent ) ) {
+				$status[ 'sent' ][ $module->name() ] = $sent;
+			}
+		}
+
+		return $status;
 	}
 
 	public function clear_status() {
-		delete_option( self::STATUS_OPTION );
+		$prefix = self::STATUS_OPTION_PREFIX;
+		delete_option( "{$prefix}_started" );
+		delete_option( "{$prefix}_queue_finished" );
+		delete_option( "{$prefix}_sent_started" );
+		delete_option( "{$prefix}_finished" );
+
+		foreach ( Jetpack_Sync_Modules::get_modules() as $module ) {
+			delete_option( "{$prefix}_{$module->name()}_queued" );
+			delete_option( "{$prefix}_{$module->name()}_sent" );
+		}
+	}
+
+	private function update_status_option( $name, $value ) {
+		$prefix = self::STATUS_OPTION_PREFIX;
+		update_option( "{$prefix}_{$name}", $value, false );
```

Comment:
This will cause a fatal error for all user using php version 4.2 or less since `update_option()` only used 2 arguments. Users should be running the latest version. 4.2 was released last year. (April 2015) but I am sure some sites are still running it.